### PR TITLE
421/show comma separators in income input field

### PR DESCRIPTION
**Pull request by KrissDrawing**
# Pull Request Template

## Issue Overview

This PR addresses #421

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

Currency field was of type number i changed it to text, so it can display commas. We store it as a text, so onSubmit there is transformation to number, also on focus it will change to number so we can modify it.
onChange there is performed checks if input is a valid number.

## How Can This Be Tested/Reviewed?

You need to go through application form to income section where will be input field.
or go to http://localhost:3000/applications/financial/income (without previous steps it will be impossible to submit).

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

**Review by emilyjablonski — CHANGES_REQUESTED** [review]
Functionally, the UX on the frontend looks great. At the moment though I'm not able to submit an application (this is surfaced in the failing Cypress suite as well in the `submit-application` test). The error we get on the backend is `income must be a string`. It looks like we are storing `income` as a string on the application entity. Honestly I'm not sure why, it seems like we should refactor this to be a number, but I would leave that for a separate ticket and save it as a string right now as we are currently doing.

**Review by emilyjablonski — APPROVED** [review]
LGTM!

File: ui-components/src/forms/Field.tsx [hunk 1]
```diff
@@ -57,20 +57,42 @@ const Field = (props: FieldProps) => {
...
+      props.setValue(props.name, e.target.value.replace(/[a-z]|[A-Z]/g, "").match(/^\d*\.?\d?\d?/g))
+    }
+  }
+
   let inputProps = { ...props.inputProps }
-  if (props.type === "currency") inputProps = { ...inputProps, step: 0.01, onBlur: formatValue }
+  if (props.type === "currency") {
+    inputProps = {
+      ...inputProps,
+      step: 0.01,
```
Comment (ludtkemorgan) [inline ui-components/src/forms/Field.tsx:88]: I'm not noticing a step functionality in the field. I know you didn't add this, but curious if it is needed anymore?

**Review by ludtkemorgan — APPROVED** [review]
Looks good!

**Pull request merged**